File: src/main/java/com/eager/questioncloud/verification/template/EmailVerificationTemplateCreator.java
package com.eager.questioncloud.verification.template;

import com.eager.questioncloud.exception.CustomException;
import com.eager.questioncloud.exception.Error;
import com.eager.questioncloud.verification.vo.EmailVerificationType;

public class EmailVerificationTemplateCreator {
    public static EmailVerificationTemplate getTemplate(EmailVerificationType emailVerificationType, String token) {
        switch (emailVerificationType) {
            case CreateUser -> {
                return new CreateUserEmailVerificationTemplate(token);
            }
            case ChangePassword -> {
                return new ChangePasswordEmailVerificationTemplate(token);
            }
            default -> {
                throw new CustomException(Error.INTERNAL_SERVER_ERROR);
            }
        }
    }
}

File: qc-domain/qc-user/src/main/kotlin/com/eager/questioncloud/verification/template/EmailVerificationTemplateCreator.kt
package com.eager.questioncloud.verification.template

import com.eager.questioncloud.verification.enums.EmailVerificationType

object EmailVerificationTemplateCreator {
    fun getTemplate(
        emailVerificationType: EmailVerificationType,
        token: String
    ): EmailVerificationTemplate {
        return when (emailVerificationType) {
            EmailVerificationType.CreateUser -> {
                CreateUserEmailVerificationTemplate(token)
            }

            EmailVerificationType.ChangePassword -> {
                ChangePasswordEmailVerificationTemplate(token)
            }
        }
    }
}
